code/part-one/validation.js

```diff
@@ -23,11 +23,28 @@ const isValidBlock = block => {
 };
 
 const isValidChain = blockchain => {
+  const { blocks } = blockchain;
 
+  if (blocks[0].previousHash !== null) {
+    return false;
+  }
+
+  if (blocks.slice(1).some((b, i) => b.previousHash !== blocks[i].hash)) {
+    return false;
+  }
+
+  if (blocks.some(b => !isValidBlock(b))) {
+    return false;
+  }
+
+  return blocks
+    .map(b => b.transactions)
+    .reduce((flat, txns) => flat.concat(txns), [])
+    .every(t => isValidTransaction(t));
 };
 
 const breakChain = blockchain => {
-
+  blockchain.blocks[1].transactions[0].amount = 1000000000;
```

Comment:
😂

Reply:
are we worried about this method failing if blockchain.addBlock() has yet to be invoked (and therefore blocks[1] === undefined)?

Reply:
I think it's fine since this will only ever be used in the context of the tests and is not intended to be particularly robust.

Reply:
👍 sounds good!